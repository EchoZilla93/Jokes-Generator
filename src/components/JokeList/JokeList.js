import React, { Fragment } from 'react';
import { useSelector } from 'react-redux';

import JokeCard from '../JokeCard/JokeCard';

export const JokeList = () => {
    const jokes = useSelector(state => state.jokes.jokes);

    return (
        <Fragment>
            {jokes &&
                jokes.map(joke => (
                    <JokeCard
                        key={joke.id}
                        id={joke.id}
                        joke={joke.value}
                        updatedAt={joke.updated_at}
                        category={joke.categories}
                        favorite={joke.favorite}
                        url={joke.url}
                    />
                ))}
        </Fragment>
    )
}

export default JokeList;