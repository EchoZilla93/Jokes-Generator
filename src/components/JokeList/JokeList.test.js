import React from 'react';
import { shallow } from 'enzyme';
import * as reactRedux from 'react-redux';
import JokeList from './JokeList';

describe('component', () => {
    it('renders component without jokes', () => {
        jest.spyOn(reactRedux, 'useSelector').mockImplementation(cb => cb({
            jokes: {
                jokes: []
            }
        }))
        const wrapper = shallow(<JokeList />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders component with jokes', () => {
        jest.spyOn(reactRedux, 'useSelector').mockImplementation(cb => cb({
            jokes: {
                jokes: [{
                    categories: ['animal'],
                    created_at: '2020-01-05 13:42:20.568859',
                    icon_url: 'https://assets.chucknorris.host/img/avatar/chuck-norris.png',
                    id: 'kv95RRh0SCie8_8q4B7hRA',
                    updated_at: '2020-01-05 13:42:20.568859',
                    url: 'https://api.chucknorris.io/jokes/kv95RRh0SCie8_8q4B7hRA',
                    value: 'The sun doesnt shine on Chuck Norris. If it did well you know when a solar eclipse is coming.'
                }]
            }
        }))
        const wrapper = shallow(<JokeList />);
        expect(wrapper).toMatchSnapshot();
    });
})