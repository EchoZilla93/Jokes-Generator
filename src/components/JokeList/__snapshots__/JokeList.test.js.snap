// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders component with jokes 1`] = `
<Fragment>
  <JokeCard
    category={
      Array [
        "animal",
      ]
    }
    id="kv95RRh0SCie8_8q4B7hRA"
    joke="The sun doesnt shine on Chuck Norris. If it did well you know when a solar eclipse is coming."
    key="kv95RRh0SCie8_8q4B7hRA"
    updatedAt="2020-01-05 13:42:20.568859"
    url="https://api.chucknorris.io/jokes/kv95RRh0SCie8_8q4B7hRA"
  />
</Fragment>
`;

exports[`component renders component without jokes 1`] = `<Fragment />`;
